include/cache_guttering.h

```diff
@@ -21,32 +21,29 @@ class CacheGuttering : public GutteringSystem {
   static constexpr size_t l1_cache_size   = 32768;    // l1 cache bytes per cpu
   static constexpr size_t l2_cache_size   = 1048576;  // l2 cache bytes per cpu
   static constexpr size_t l3_cache_size   = 33554432; // l3 cache bytes in total
+
+  static constexpr size_t cache_sizes[3] = {32768, 1048576, 33554432};
   static constexpr size_t cache_line      = 64; // number of bytes in a cache_line
-  static constexpr size_t cache_bytes_per_child = 2 * cache_line;
-  static constexpr size_t RAM_bytes_per_child   = 8 * cache_line;
-
-  // basic 'tree' params, hardcoded for now. TODO: Determine by sizes later
-  static constexpr size_t fanout          = 32;  // fanout for cache gutters. Must be power of 2
-  static constexpr size_t num_l1_bufs     = 8;   // number l1 buffers. Must be power of 2
-  static constexpr size_t num_l2_bufs     = num_l1_bufs * fanout;
-  static constexpr size_t num_l3_bufs     = num_l2_bufs * fanout;
-  static constexpr size_t max_RAM1_bufs   = num_l3_bufs * fanout;
-  static constexpr size_t l1l2buffer_elms = cache_bytes_per_child * fanout / sizeof(update_t);
-  static constexpr size_t l3buffer_elms   = cache_bytes_per_child * fanout / sizeof(update_t);
-
-  // bit length variables
-  static constexpr int l1_bits   = log2_constexpr(num_l1_bufs);
-  static constexpr int l2_bits   = log2_constexpr(num_l2_bufs);
-  static constexpr int l3_bits   = log2_constexpr(num_l3_bufs);
-  static constexpr int RAM1_bits = log2_constexpr(max_RAM1_bufs);
+  static constexpr size_t numerator         = 1;
+  static constexpr size_t divisor         = 1;
```

Comment:
What exactly are these variables for?

Reply:
The original commit allocated 16 times as much memory, although my original plan was to allocate less memory than the cache. This allows us to over or under allocate memory relative to our cache size. It could probably just be a double.

Reply:
From the math I just did it looks like we did over use the cache in commit [275a5ce](https://github.com/GraphStreamingProject/GutterTree/pull/57/commits/275a5ce652b008b13cce516c5f44db3d9ae6a183). However, I'm getting that we use twice as much cache as we should have. 

For example, for L1, fanout of 16 with 1 cache line per child give buffer size of 1 KiB. With 64 l1 buffers that's 64 KiB which is twice the L1 cache size of 32KiB. Am I doing something wrong?

Reply:
The sanity check I did was sizeof(L!_gutters) / L!_cache_size. I got 16. Double check the math as implemented in [275a5ce](https://github.com/GraphStreamingProject/GutterTree/pull/57/commits/275a5ce652b008b13cce516c5f44db3d9ae6a183), we definitely didn't implement what we intended.

Reply:
Yeah turns out we also didn't divide by sizeof(update_t) for num_buffer_elms